website/src/components/Chat/ChatConversation.tsx

```diff
@@ -29,35 +35,34 @@ export const ChatConversation = ({ chatId }: ChatConversationProps) => {
   const [queueInfo, setQueueInfo] = useState<QueueInfo | null>(null);
   const [isSending, setIsSending] = useBoolean();
 
-  useSWR<ChatItem>(API_ROUTES.GET_CHAT(chatId), get, { onSuccess: (chat) => setMessages(chat.messages) });
-  const { getValues } = useFormContext<WorkParametersInput>();
+  useSWR<ChatItem>(API_ROUTES.GET_CHAT(chatId), get, {
+    onSuccess: (chat) =>
+      setMessages(chat.messages.sort((a, b) => new Date(a.created_at).getTime() - new Date(b.created_at).getTime())),
+  });
+  const { getValues: getFormValues } = useFormContext<ChatConfigForm>();
   const send = useCallback(async () => {
     const content = inputRef.current?.value.trim();
     if (!content || !chatId) {
       return;
     }
     setIsSending.on();
 
-    // find last assistant message, is usually last message but not always
-    // TODO: the inference server does not return the messages sorted, this is only a best guess
+    // find last VALID assistant message
     const parent_id =
       messages
         .slice()
         .reverse()
-        .find((m) => m.role === "assistant")?.id ?? null;
-    const params = getValues();
-
-    const response: InferencePostMessageResponse = await post(API_ROUTES.CREATE_CHAT_MESSAGE, {
-      arg: {
-        chat_id: chatId,
-        content,
-        parent_id,
-        work_parameters: {
-          ...params,
-          top_k: params.top_k === null ? null : Math.floor(Math.pow(32000, params.top_k)),
```

Comment:
Have you tested the top_k here? Andreas suggest an exponential function for this.

Reply:
ah nvm I see the config below

Reply:
Yeah just for info: this works for the custom config where the slider goes between 0 and 1, but for all the presets it computes `pow(32000, 50)` or similar.

I did not find a clean way of changing this so I changed min and max. Arguably, no more than 1000 is needed, because the sampling probability is already so low it might as well be zero.